File: DMS-main/DMS_API/DemandMgmt.Business/DataBase/Demand.cs
using System;
using System.Collections.Generic;

namespace DemandMgmt.Business.DataBase;

public partial class Demand
{
    public int Id { get; set; }

    public string? Uid { get; set; }

    public string? Asset { get; set; }

    public int? Role { get; set; }

    public string? RoleDetails { get; set; }

    public string? YearsOfExp { get; set; }

    public int Department { get; set; }

    public string? RaisedBy { get; set; }

    public int PrimarySkills { get; set; }

    public string? SecondarySkills { get; set; }

    public string? SkillDetails { get; set; }

    public string? RequiredByDate { get; set; }

    public string? RaisedOn { get; set; }

    public string? KtstartDate { get; set; }

    public string? KtendDate { get; set; }

    public int? HlcApproval { get; set; }

    public string? HlcApprovalDate { get; set; }

    public string? JdFileLocation { get; set; }

    public string? DemandInitiationFileLocation { get; set; }

    public string? ContractorReplacement { get; set; }

    public string? OtherContractor { get; set; }

    public string? RechargeDate { get; set; }

    public int? MtbShare { get; set; }

    public int? TtbShare { get; set; }

    public int Status { get; set; }

    public string? AdditionalRemark { get; set; }

    public int? IsActive { get; set; }

    public int? Practice { get; set; }

    public virtual PracticeMaster? PracticeNavigation { get; set; }
}

File: DMS-main/DMS_UI/src/Domain/Demand/Demand.ts
import { DataRouteMatch } from "react-router-dom";

export interface Demand {
    id: number;
    uid: string;
    asset: string;
    role: number;
    roleDetails: string;
    yearsOfExp: number;
    department: number;
    practice: number;
    raisedBy: string;
    primarySkills: number;
    secondarySkills: string[];
    skillDetails: string;
    requiredByDate: string | null;
    raisedOn: string | null;
    kTstartDate: string;
    kTendDate: string;
    hlcApproval: boolean;
    hlcApprovalDate: string | null;
    jdFileLocation: string;
    demandInitiationFileLocation: string;
    contractorReplacement: number;
    otherContractor: string;
    rechargeDate: string | null;
    mtbShare: number;
    tTBShare: number;
    status: number;
    additionalRemark: string;
    isActive: boolean;
}



export enum DemandStatus
{
    Initiated,
    DemandManagerAprovalPending,
    DemandManagerAprovalRejected,
    DemandApprover_PH_ApprovalPending,
    DemandApprover_PH_ApprovalRejected,
    DemandApprover_DSH_ApprovalPending,
    DemandApprover_DSH_ApprovalRejected,
    SupplyManagerApprovalPending,
    SupplyManagerApprovalRejected,
    SupplyApproverApprovalPending,
    SupplyApproverApprovalRejected,
    SupplyRecruiterApprovalPending,
    SupplyRecruiterApprovalRejected,
    Fulfilled
}